docs/maui/layouts/DockLayout.md

```diff
@@ -0,0 +1,122 @@
+
+### C#
+
+To set the docking position from C#, use `DockLayout.SetDockPosition(IView, DockPosition)` to apply the attached `DockPosition` property.
+
+```csharp
+var button = new Button { Text = "Top", HeightRequest = 50 };
+DockLayout.SetDockPosition(button, DockPosition.Top);
+```
+
+Alternatively, `DockLayout` contains a fluent-style `Add(IView, DockPosition)` method that adds a child view to the layout container and sets it's docking position.
+
+```csharp
+var dockLayout = new DockLayout();
+var button = dockLayout.Add(new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top);
+```
+
+Using the overload, a `DockLayout` can be constructed conveniently in C# as shown here:
+
+```csharp
+using CommunityToolkit.Maui.Layouts;
+
+var page = new ContentPage
+{
+	Content = new DockLayout
+	{
+		{ new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top },
+		{ new Button { Text = "Bottom", HeightRequest = 70 }, DockPosition.Bottom },
+		{ new Button { Text = "Left", WidthRequest = 80 }, DockPosition.Left },
+		{ new Button { Text = "Right", WidthRequest = 90 }, DockPosition.Right },
+        { new Button { Text = "Center" } },
+	}
+};
+```
+
+Note: `DockPosition.None` is the default and can be omitted.
+
+## Customizing a DockLayout
+
+A `DockLayout` container supports arbitrary `Padding` as well as several `DockLayout`-specific properties for customization. An example in XAML with all available options is given here:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <mct:DockLayout HeightRequest="400"
+                    WidthRequest="600"
+                    Padding="10,20,30,40"
+                    VerticalSpacing="10"
+                    HorizontalSpacing="15"
+                    ShouldExpandLastChild="False">
+        ...
+    </mct:DockLayout>
+    
+</ContentPage>
+```
+
+## Properties
+
+|Property  |Type  |Description  |
+|---------|---------|---------|
+| `Padding` | `Thickness` | Gets or sets the padding around the layout container (inherited from `Layout`). |
+| `VerticalSpacing` | `double` | Gets or sets the **vertical** spacing between docked views. |
+| `HorizontalSpacing` | `double` | Gets or sets the **horizontal** spacing between docked views. |
+| `ShouldExpandLastChild` | `bool` | If true, the last child is expanded to fill the remaining space (default: `true`). |
+
+`VerticalSpacing` and `HorizontalSpacing` is applied between neighboring views in the `DockLayout`. For example, `HorizontalSpacing` is added between *Left*, *None*, and *Right* views, but also between neighboring views in the same `DockPosition` such as multiple views docked to the *Left*. `VerticalSpacing` is rendered between vertically stacked views in *Top*, *None*, and *Bottom* positions.
```

Comment:
I wonder if this information would be better included in the table above with the properties they directly relate to. What do you think?

Reply:
Feel free to relocate this section as you see fit. I like to keep (such) tables simple, so that you can get an overview of all properties at one glance. And better put detailed explanations below. But that's certainly a personal preference.

Reply:
Ok, I played a little with the table and tried to inline details on Horizontal/VerticalSpacing. Last (and separate) commit. Can be reverted easily, if you dont like it.
But as I said, feel free to change as you like best ...

Reply:
Awesome thanks, as per my latest comment, let's see what Gerald notes on the preview of the actual docs page to make sure it handles this formatting.